options/path.c

```diff
@@ -315,12 +316,15 @@ bool mp_is_url(bstr path)
     int proto = bstr_find0(path, "://");
     if (proto < 1)
         return false;
-    // The protocol part must be alphanumeric, otherwise it's not an URL.
+    // Per RFC3986, the first character of the protocol must be alphanumeric, 
+    // the rest must be alphanumeric plus - + and .
     for (int i = 0; i < proto; i++) {
         unsigned char c = path.start[i];
-        if (!(c >= 'a' && c <= 'z') && !(c >= 'A' && c <= 'Z') &&
-            !(c >= '0' && c <= '9') && c != '_')
+        if ((i == 0 && !mp_isalpha(c)) ||
```

Comment:
* The comment says:
> ... must be alphanumeric
* The check is:
> ...must be from alphabet

I think that if tmm1 passed this, this is most likely meant to be like this (first character has to be alphabetic), and if so this comment can be easily fixed when being merged&pushed.

Reply:
According to https://tools.ietf.org/html/rfc3986#section-3.1 yes, the scheme has to be `ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )`